test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/SignerAndEndpointOverridesTest.java

```diff
@@ -0,0 +1,228 @@
+    //  the V4 tests pass right now only because the V4 auth-scheme is the default for this client
+    @Disabled("Expected to fail as of now.")
+    @Test
+    public void test_whenV4aEndpointAuthSchemeWithSignerOverride_thenEndpointParamsShouldPropagateToSigner() {
+        ProtocolQueryClient client = ProtocolQueryClient
+            .builder()
+            .authSchemeProvider(v4aAuthSchemeProviderOverride())
+            .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("akid", "skid")))
+            .endpointProvider(v4aEndpointProviderOverride())
+            .region(Region.US_EAST_1)
+            .overrideConfiguration(o -> o.putAdvancedOption(SIGNER, mockSigner))
+            .build();
+
+        try {
+            client.streamingInputOperation(r -> {}, RequestBody.fromString("test"));
+        } catch (Exception expected) {
+        }
+
+        verify(mockSigner).sign(
+            any(SdkHttpFullRequest.class),
+            argThat(attrs ->
+                        "us-east-1".equals(attrs.getAttribute(AwsSignerExecutionAttribute.SIGNING_REGION).id()) &&
+                        "query-test-v4a".equals(attrs.getAttribute(AwsSignerExecutionAttribute.SERVICE_SIGNING_NAME)) &&
+                        !attrs.getAttribute(AwsSignerExecutionAttribute.SIGNER_DOUBLE_URL_ENCODE))
+        );
+    }
+
+    // TODO(sra-identity-and-auth): Enable this test once an auth-scheme override is respected when resolving the auth-scheme -
+    //  the V4 tests pass right now only because the V4 auth-scheme is the default for this client
+    @Disabled("Expected to fail as of now.")
+    @Test
+    public void testAsync_whenV4aEndpointAuthSchemeWithSignerOverride_thenEndpointParamsShouldPropagateToSigner() {
+        ProtocolQueryAsyncClient client = ProtocolQueryAsyncClient
+            .builder()
+            .authSchemeProvider(v4aAuthSchemeProviderOverride())
+            .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("akid", "skid")))
+            .endpointProvider(v4aEndpointProviderOverride())
+            .region(Region.US_EAST_1)
+            .overrideConfiguration(o -> o.putAdvancedOption(SIGNER, mockSigner))
+            .build();
+
+        try {
+            client.streamingInputOperation(r -> {}, AsyncRequestBody.fromString("test")).join();
+        } catch (Exception expected) {
+        }
+
+        verify(mockSigner).sign(
+            any(SdkHttpFullRequest.class),
+            argThat(attrs ->
+                        "us-east-1".equals(attrs.getAttribute(AwsSignerExecutionAttribute.SIGNING_REGION).id()) &&
+                        "query-test-v4a".equals(attrs.getAttribute(AwsSignerExecutionAttribute.SERVICE_SIGNING_NAME)) &&
+                        !attrs.getAttribute(AwsSignerExecutionAttribute.SIGNER_DOUBLE_URL_ENCODE))
+        );
+    }
+
+    private static ProtocolQueryAuthSchemeProvider v4AuthSchemeProviderOverride() {
+        return __ -> {
```

Comment:
Nit, can we use a letter here instead? (e.g., `x -> {...}`)

Reply:
It's unused, and this is a fairly common when the variable is unused.

Reply:
Not in Java, and I can't find any other test in the project using it (didn't search thoroughly), but up to the team.

Reply:
(c.f., the commonly used pattern on some other languages `_` is not even possible in Java since a single underscore is reserved).

Reply:
Yep, hence the usage of two underscores. Changed to 'x'.

Reply:
You forgot this one and the other below.